docs/release/release-process.md

```diff
@@ -140,6 +171,29 @@ The `master` branch can now be opened for new features since we have a safe
 release branch to continue bug fixes and improvements for the release itself.
 Essentially, `master` is free to now diverge from the release branch.
 
+### Release Branch Prep
```

Comment:
Given my slip-up on opening PRs against `master` in the `v0.9` release, it would be nice if we could establish some safe guards here. I think following these new more detailed notes helps a ton, an usually it will get caught in review, but I have seen some projects handle this sort of thing with a separate tool that is used for release that basically just wraps a Github client and enforces certain policies. Out-of-scope for this document right now, but always good to improve process where possible :)

Reply:
yes i definitely agree! more automation is definitely desired from this process to not only reduce opportunities for human mistakes but also save time and energy too.  Hopefully you're getting good ideas from being in sig release upstream :)